dashboard/src/components/hackathon/HackathonAttendanceMode.vue

```diff
@@ -0,0 +1,274 @@
+    class="z-50"
+    v-model="showDialog"
+    :options="{
+      title: 'Change Mode of Attendance',
+    }"
+  >
+    <template #body-content>
+      <div
+        class="flex flex-col gap-2 items-center text-center p-3 rounded w-full bg-orange-50 text-orange-600 border-2 border-dashed border-orange-600 text-xs"
+      >
+        <svg
+          xmlns="http://www.w3.org/2000/svg"
+          width="24"
+          height="24"
+          viewBox="0 0 24 24"
+          fill="none"
+          stroke="currentColor"
+          stroke-width="2"
+          stroke-linecap="round"
+          stroke-linejoin="round"
+          class="icon icon-tabler h-6 w-6 icons-tabler-outline icon-tabler-alert-circle"
+        >
+          <path stroke="none" d="M0 0h24v24H0z" fill="none" />
+          <path d="M3 12a9 9 0 1 0 18 0a9 9 0 0 0 -18 0" />
+          <path d="M12 8v4" />
+          <path d="M12 16h.01" />
+        </svg>
+        <p>
+          <b>Attention!</b> You are about to change your attendance mode for
+          this hackathon.<br /><br />
+          Please note that requests for in-person attendance are subject to
+          approval by the organizers due to limited on-site availability.
+        </p>
+      </div>
+      <FormControl
+        label="Mode of Attendance"
+        v-model="attendanceMode"
+        type="select"
+        class="mt-4"
+        :options="attendanceModeOptions"
+      />
+      <FormControl
+        v-if="attendanceMode == 'local' || attendanceMode == 'local-pending'"
+        label="Select LocalHost"
+        :disabled="attendanceMode == 'local'"
+        v-model="selectedLocalhost"
+        type="select"
+        class="mt-4"
+        :options="
+          localhosts.data.map((localhost) => ({
+            label: `${localhost.localhost_name}`,
+            value: localhost.name,
+          }))
+        "
+      />
+      <ErrorMessage :message="errorMessage" class="mt-2" />
+    </template>
+    <template #actions>
+      <div class="grid grid-cols-2 gap-4">
+        <Button
+          label="Cancel"
+          size="sm"
+          theme="gray"
+          @click="showDialog = false"
+        />
+        <Button
+          label="Change"
+          size="sm"
+          theme="green"
+          :disabled="handleDisabled()"
+          @click="handleAttendanceModeChange"
+        />
+      </div>
+    </template>
+  </Dialog>
+  <div
+    class="flex flex-col gap-1 border-2 rounded border-gray-500 border-dashed min-w-[220px] p-4"
+  >
+    <h5 class="text-xs text-gray-700 uppercase font-medium">
+      Your Mode of attendance
+    </h5>
+    <div class="flex w-full items-center justify-between">
+      <div
+        class="text-base font-medium"
+        :class="attendanceMode == 'local-pending' ? 'text-orange-600' : ''"
+      >
+        {{
+          attendanceMode == 'remote'
+            ? 'Remote'
+            : attendanceMode == 'local-pending'
+              ? 'On-Site (Pending Request)'
+              : 'On-site'
```

Comment:
If On-site - adding the localhost name in the block would be good too.
<img width="706" alt="Screenshot 2024-07-15 at 6 35 14 PM" src="https://github.com/user-attachments/assets/dfbb384c-cb99-447a-a735-fd44c10702e7">